src/Features/LanguageServer/Protocol/ExternalAccess/VSCode/API/VSCodeTelemetryLogger.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.CodeAnalysis.Internal.Log;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.VSCode.API;
+
+/// <summary>
+/// Allows VSCode to implement a telemetry logger for <see cref="ILogger"/> events.
+/// </summary>
+internal abstract class VSCodeTelemetryLogger : ILogger
+{
+    bool ILogger.IsEnabled(FunctionId functionId)
+    {
+        return IsEnabled(functionId.ToString());
```

Comment:
❗ This is extremely expensive in practice. The map from `FunctionId`&rarr;`string` needs to be in a `Dictionary<TKey, TValue>` (or `ImmutableSegmentedDictionary<TKey, TValue>`)